src/libraries/Common/src/Interop/Windows/SspiCli/Interop.SSPI.cs

```diff
@@ -215,6 +215,97 @@ public enum Flags
             }
         }
 
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct SCH_CREDENTIALS
+        {
+            public const int CurrentVersion = 0x5;
+
+            public int dwVersion;
+            public int dwCredformat;
+            public int cCreds;
+
+            // ptr to an array of pointers
+            // There is a hack done with this field.  AcquireCredentialsHandle requires an array of
+            // certificate handles; we only ever use one.  In order to avoid pinning a one element array,
+            // we copy this value onto the stack, create a pointer on the stack to the copied value,
+            // and replace this field with the pointer, during the call to AcquireCredentialsHandle.
+            // Then we fix it up afterwards.  Fine as long as all the SSPI credentials are not
+            // supposed to be threadsafe.
+            public IntPtr paCred;
```

Comment:
Why are some things declared as actual strongly-typed pointers and others like this declared just as IntPtr?

Reply:
I left same as existing SCHANNEL_CRED in most cases. 
The notable difference is SCH_CREDENTIALS* and that was done to pass reference via RunImpersonated() lambda.

Reply:
In that case we get opaque pointer and we are it in safe handle. I updated the structure to use native pointer in places where caller (WindowesPAL) is responsible for pinning the data. 
In this particular case I also made update to construct expected layout so there is no trickery in SecuritySafeHandle. I think this will be handy soon to pass entire certificate chain. I'll do more testing with chains and I'll fix the leafy code when that gets ready.